### ISSUE TITLE
@interact missing argument labels in JupyterLab

### ISSUE THREAD

**Issue by jasongrout** [ipython/ipywidgets#670]
The sliders below should be labeled with `x` and `y`, the argument names.

```
from ipywidgets import interact, Output, HBox
left = Output()
right = Output()

with left:
    @interact(x=(0,10))
    def f(x):
        @interact(y=(0,x))
        def g(y):
            with right:
                print('Product',x*y)

print("Product calculator")                
HBox([left, right])
```

*Edit: This only seems to be a problem in JupyterLab, not the notebook*

**Comment by jdemeyer** [comment]
Works for me:
![sliders](https://cloud.githubusercontent.com/assets/3625785/18675358/6d1875e6-7f52-11e6-9916-435a203c0ea9.png)

**Comment by gnestor** [comment]
I can also verify that it's working on notebook master and ipywidgets master.

![image](https://cloud.githubusercontent.com/assets/512354/18959759/31112c70-861c-11e6-84b9-e9da2533e0c6.png)

**Comment by jasongrout** [comment]
It's not working for me in JupyterLab, but it is working for me in the notebook.

**Comment by jasongrout** [comment]
It appears that widgets inside a jupyterlab Output widget do not trigger the onAttached method, so the displayed promise never gets resolved. Here's a simpler example that doesn't work:

```python
from IPython.display import display
from ipywidgets import Output, IntSlider
out = Output()

with out:
    display(IntSlider(description="hi"))
display(out)
```

**Comment by jasongrout** [comment]
Fixed in #889.
